app/assets/javascripts/user_form.js

```diff
@@ -7,6 +7,29 @@ $(function () {
      $('.form-group.user_company_info').toggle(this.checked);
   }).change(); //ensure visible state matches initially
 
+
+   //update coach availability
+
+   $(function () {
+      var $update_availability = $('[data-behavior="update_availability"]');
+      $update_availability.on('click',function(){
+        var url =$(this).data("url");
+        updateAvailability(url);
+      });
+    });
+
+   function updateAvailability(url) {
+      $.ajax({
+        type:     'PUT',
```

Comment:
Also: I think it's the wrong key. It should be 
```js
method: "PUT"
```
since this is about type of the request.

Maybe you were referring to `dataType`??? It's optional here, jQuery will do an intelligent guess, but you could also deliberately set it to whatever you need. Would be
```js
dataType: "script"
```
here I suppose.